clients/githubrepo/contributors.go

```diff
@@ -42,50 +56,170 @@ func (handler *contributorsHandler) init(ctx context.Context, repourl *Repo) {
 	handler.contributors = nil
 }
 
-func (handler *contributorsHandler) setup() error {
+func (handler *contributorsHandler) setup(fileReader io.ReadCloser) error {
 	handler.once.Do(func() {
+		defer fileReader.Close()
 		if !strings.EqualFold(handler.repourl.commitSHA, clients.HeadSHA) {
 			handler.errSetup = fmt.Errorf("%w: ListContributors only supported for HEAD queries", clients.ErrUnsupportedFeature)
 			return
 		}
-		contribs, _, err := handler.ghClient.Repositories.ListContributors(
-			handler.ctx, handler.repourl.owner, handler.repourl.repo, &github.ListContributorsOptions{})
+
+		handler.contributors = append(handler.contributors, getContributors(handler)...)
+		handler.contributors = append(handler.contributors, getCodeOwners(handler, fileReader)...)
+
+		handler.errSetup = nil
+	})
+	return handler.errSetup
+}
+
+func getContributors(handler *contributorsHandler) []clients.User {
+	contribs, _, err := handler.ghClient.Repositories.ListContributors(
+		handler.ctx, handler.repourl.owner, handler.repourl.repo, &github.ListContributorsOptions{})
```

Comment:
i'll check later when I have my IDE which highlights moved code better than GitHub's UI does. but this is the existing logic?